spec/predicates/human-review-crev.md

```diff
@@ -0,0 +1,161 @@
+# Predicate type: Dependency Reviews (crev)
+
+Type URI: (tentative) https://in-toto.io/attestation/human-review/crev/v0.1
+
+Note: At the time of writing, the crev project uses a separate versioning
+scheme, with the current version being -1. In future, this predicate may adopt
+a URI associated with that project.
+
+Version: v0.1.0
+
+## Purpose
+
+This attestation type is used to describe the results of human review of
+dependency source code. The format is based on the
+[crev project](https://github.com/crev-dev/crev).
+
+## Use Cases
+
+As noted above, crev enables social review of popular open source software
+dependencies. A crev review includes information such as the thoroughness of
+the review, understanding of the source code, and a final rating.
+
+## Model
+
+Most modern software have external dependencies. Dependency review is the
+process of reviewing and verifying the source code of a particular
+dependency, and can be performed by one or more of several actors in the supply
+chain. The developer importing a new dependency can perform the review or a
+dedicated security team can be tasked with it.
+
+## Schema
+
+```json
+{
+    "_type": "https://in-toto.io/Statement/v1",
+    "subject": [{...}],
+    "predicateType": "https://in-toto.io/attestation/human-review/crev/v0.1",
+    "predicate": {
+        "result": "positive|negative",
+        "reviewer": {
+            "idType": "crev",
+            "id": "<ID>",
+            "url": "<URL>"        
+        },
+        "reviewTime": "<TIMESTAMP>",
+        "thoroughness": "high|medium|low",
+        "understanding": "high|medium|low",
+        "comment": "<STRING>"
+    }
+}
+```
+
+### Parsing Rules
+
+This predicate follows the
+[in-toto Attestation Framework's parsing rules](../v1/README.md#parsing-rules).
+
+### Fields
+
+The subject of this predicate type is a specific package and its version in some
+ecosystem.
+
+`result`, _enum_, _required_
+
+Specifies if the overall rating of the dependency is `positive` or `negative`.
+
+`reviewer` _object_, _required_
+
+Identifies the reviewer. This has some meaning for crev's trust proliferation
+aspects, but the identity of the reviewer can also be mapped based on in-toto's
+functionary handling. `idType` is used to determine the contents of `reviewer`.
+The `url` is a reference to the reviewer's crev-proofs repository.
```

Comment:
A question and a suggestion. Q: What are possible `idType` values, and who determines these? Rec: I'd add in here that this field is intended to correspond to the `from` field in the Code Review Proofs format when the `idType` matches the ID for Crev.

Reply:
There's a bit of a disconnect here, because the crev [code review](https://github.com/crev-dev/cargo-crev/blob/master/crev-lib/rc/doc/editing-code-review.md) type includes `from` but missing from the [package review](https://github.com/crev-dev/cargo-crev/blob/master/crev-lib/rc/doc/editing-package-review.md) type. Same for `date`. I suspect `date` may be omitted as a package-wide review is meant as an ongoing document with updates as new versions / advisories emerge. On the other hand, I'm not sure about the `from` field. I wonder if in the in-toto context we want to drop it and lean entirely on the signature. That takes us to a broader conversation about how in-toto supports crev, though.